woocommerce/woocommerce-ios
# Coupons: Update list UI

Closes: #5764 


### Description

This PR updates UI of the coupon items on the Coupons screen: 
- Add a new table view cell - which looks similar to the Order list cell but I added a more generic one to be reused later if needed.
- Calculate expiry status based on the expiry date of the coupon
- Display discount type and expiry status on the coupon cells. There might be an update to the subtitle of the cell though since the UI hasn't been finalized yet; I'll add another PR for the update later 😅.

Some extra bits:
- Remove filter for non-default coupons following the internal discussion: p91TBi-7bG-p2#comment-8153
- Update empty state view with new copies and images.

### Testing instructions

- Switch to a store that doesn't have any coupon set up.
- Select Menu tab > Coupons. Notice that the empty view matches with design: O8MT8FHx8N2zP9pc2namgl-fi-305%3A3567
- Switch to a store that has coupons and at least one with a non-default discount type.
- Notice that the displayed details on the cells are correct; and no coupons are filtered out.

### Screenshots

| Coupon List | Empty Light Mode | Empty Dark Mode |
| ----- | ----- | ----- |
| <img src="https://user-images.githubusercontent.com/5533851/150472009-f3d33dee-a188-4f13-872a-8e482c519660.png" width=320 /> |  <img src="https://user-images.githubusercontent.com/5533851/150472057-11e0106a-3ea1-429e-9b44-adbe1435bd6a.png" width=320 /> | <img src="https://user-images.githubusercontent.com/5533851/150472112-729f3039-072c-4e72-a915-71543c8b1535.png" width=320 /> |

---
